frontend/src/scenes/notebooks/Nodes/NotebookNodeLink.tsx

```diff
@@ -38,26 +42,32 @@ const ICON_MAP = {
 }
 
 const Component = (props: NodeViewProps): JSX.Element => {
+    const { shortId } = useValues(notebookLogic)
+    const { notebookSideBarShown } = useValues(notebookSidebarLogic)
+    const { setNotebookSideBarShown, selectNotebook } = useActions(notebookSidebarLogic)
+
     const href: string = props.node.attrs.href
 
-    const [path, icon] = useMemo(() => {
+    const [path, pathStart, internal] = useMemo(() => {
         const path = href.replace(window.location.origin, '')
         const pathStart = path.split('/')[1]?.toLowerCase()
+        const internal = href.startsWith(window.location.origin)
 
-        return [path, ICON_MAP[pathStart] || <IconLink />]
+        return [path, pathStart, internal]
     }, [href])
 
+    const handleOnClick = useCallback(() => {
+        if (internal && !notebookSideBarShown) {
+            selectNotebook(shortId)
+            setNotebookSideBarShown(true)
+        }
+    }, [internal, shortId, setNotebookSideBarShown, selectNotebook])
```

Comment:
we should probably move some or all of this to the notebookSidebarLogic

A selector on that logic that returns a function that takes `internal` and `shortId` as args would probably be a step.

The selector would be equivalent to `useCallback` I think because it also only changes when its dependencies change (with the difference being that `internal` is provided as a simple first step

Reply:
Thanks for the suggestion! My Kea isn't brilliant but [I think I implemented](https://github.com/PostHog/posthog/pull/16388/files#diff-1dee0d5b16ce2d9fa66f10827c0bd5aab8b6ff244b527cdd5c9d119b599f40edR62) what you suggested and tested that everything still works.

One question I had around Kea as I went was whether there would be a difference between:
```
selectors(({ actions }) => ({
    notebookLinkClicked: [
        (s) => [s.notebookSideBarShown],
        (notebookSideBarShown): ((shortId: string, internal: boolean) => void) => {
            return (shortId: string, internal: boolean) => {
                if (!notebookSideBarShown && internal) {
                    actions.selectNotebook(shortId)
                    actions.setNotebookSideBarShown(true)
                }
            }
       },
    ],
})),
```

and

```
selectors(({ values, actions }) => ({
    notebookLinkClicked: [
        () => [],
        (): ((shortId: string, internal: boolean) => void) => {
            return (shortId: string, internal: boolean) => {
                if (!values.notebookSideBarShown && internal) {
                    actions.selectNotebook(shortId)
                    actions.setNotebookSideBarShown(true)
                }
            }
       },
    ],
})),
```

They both seem to work. I read that a selector is created for each value - is there a preference for referencing the value over the selector in any case?